external/esp_idf_port/esp_lwip/dhcpserver.c

```diff
@@ -0,0 +1,1362 @@
+#define DHCP_HTYPE_ETHERNET 1
+#define DHCP_HLEN_ETHERNET  6
+#define DHCP_MSG_LEN      236
+
+#define DHCPS_SERVER_PORT  67
+#define DHCPS_CLIENT_PORT  68
+
+#define DHCPDISCOVER  1
+#define DHCPOFFER     2
+#define DHCPREQUEST   3
+#define DHCPDECLINE   4
+#define DHCPACK       5
+#define DHCPNAK       6
+#define DHCPRELEASE   7
+
+#define DHCP_OPTION_SUBNET_MASK   1
+#define DHCP_OPTION_ROUTER        3
+#define DHCP_OPTION_DNS_SERVER    6
+#define DHCP_OPTION_REQ_IPADDR   50
+#define DHCP_OPTION_LEASE_TIME   51
+#define DHCP_OPTION_MSG_TYPE     53
+#define DHCP_OPTION_SERVER_ID    54
+#define DHCP_OPTION_INTERFACE_MTU 26
+#define DHCP_OPTION_PERFORM_ROUTER_DISCOVERY 31
+#define DHCP_OPTION_BROADCAST_ADDRESS 28
+#define DHCP_OPTION_REQ_LIST     55
+#define DHCP_OPTION_END         255
+
+//#define USE_CLASS_B_NET 1
+#define DHCPS_DEBUG          0
+#define DHCPS_LOG printf
+
+#define MAX_STATION_NUM CONFIG_LWIP_DHCPS_MAX_STATION_NUM
+
+#define DHCPS_STATE_OFFER 1
+#define DHCPS_STATE_DECLINE 2
+#define DHCPS_STATE_ACK 3
+#define DHCPS_STATE_NAK 4
+#define DHCPS_STATE_IDLE 5
+#define DHCPS_STATE_RELEASE 6
+
+struct _dhcps_handle {
+	struct netif *interface;
+	struct udp_pcb *dhcps_pcb;
+} dhcps_handle[TCPIP_ADAPTER_IF_MAX];
+static int netifnum = 0;
+
+typedef struct _list_node {
+	void *pnode;
+	struct _list_node *pnext;
+} list_node;
+
+////////////////////////////////////////////////////////////////////////////////////
+
+static const u32_t magic_cookie = 0x63538263;
+
+static struct udp_pcb *pcb_dhcps = NULL;
+static ip4_addr_t broadcast_dhcps;
+static ip4_addr_t server_address;
+static ip4_addr_t dns_server = { 0 };
+
+static ip4_addr_t client_address;	//added
+static ip4_addr_t client_address_plus;
+
+static list_node *plist = NULL;
+static bool renew = false;
+
+static dhcps_lease_t dhcps_poll;
+static dhcps_time_t dhcps_lease_time = DHCPS_LEASE_TIME_DEF;	//minute
+static dhcps_offer_t dhcps_offer = 0xFF;
+static dhcps_offer_t dhcps_dns = 0x00;
+static dhcps_cb_t dhcps_cb;
+static void add_netif_dhcps(struct netif *iface, struct udp_pcb *dhcps_pcb);
+static struct udp_pcb *get_netif_dhcps(struct netif *iface);
+static void reset_netif_dhcps(struct netif *iface);
+/******************************************************************************
+ * FunctionName : dhcps_option_info
+ * Description  : get the DHCP message option info
+ * Parameters   : op_id -- DHCP message option id
+ *                opt_len -- DHCP message option length
+ * Returns      : DHCP message option addr
+*******************************************************************************/
+static void add_netif_dhcps(struct netif *iface, struct udp_pcb *dhcps_pcb)
+{
+	dhcps_handle[netifnum].interface = iface;
+	dhcps_handle[netifnum++].dhcps_pcb = dhcps_pcb;
+	if (netifnum >= TCPIP_ADAPTER_IF_MAX) {
+		DHCPS_LOG("dhcps: overflow\n");
+	}
+}
+
+static struct udp_pcb *get_netif_dhcps(struct netif *iface)
+{
+	for (int i = 0; i < TCPIP_ADAPTER_IF_MAX; i++) {
```

Comment:
it's better to use int i; for(i=0;....).

Reply:
done